### PR TITLE
feat(swarm): remove unused types from public API

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/CHANGELOG.md [hunk 1]
```diff
@@ -84,6 +79,7 @@
 [PR 3373]: https://github.com/libp2p/rust-libp2p/pull/3373
 [PR 3374]: https://github.com/libp2p/rust-libp2p/pull/3374
 [PR 3375]: https://github.com/libp2p/rust-libp2p/pull/3375
+[PR XXXX]: https://github.com/libp2p/rust-libp2p/pull/XXXX
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
[PR 3497]: https://github.com/libp2p/rust-libp2p/pull/3497
```

File: swarm/CHANGELOG.md [hunk 2]
```diff
@@ -69,6 +60,10 @@
 - Introduce `ListenError` and use it within `SwarmEvent::IncomingConnectionError`.
   See [PR 3375].
 
+- Remove `PendingConnectionError`, `PendingInboundConnectionError` and `PendingOutboundConnectionError` from the public API.
+  They are no longer referenced anywhere with the addition of `ListenError`.
+  See [PR XXXX].
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
  See [PR 3497].
```

**Review by mxinden — APPROVED** [review]

**Comment by thomaseizinger** [comment]
@mxinden I assume your approval also means that you are okay with editing the changelog. I figured that removing intermediate entries is good to not confuse users unecessarily.

**Comment by mxinden** [comment]
I am in favor of the changelog edits introduced here :+1: thanks for asking.

**Pull request merged**